### PR TITLE
[Blocks Editor]: Can’t create a block below an image or code block

### PR DESCRIPTION
**Pull request by madhurisandbhor**
### What does it do?

Adding a new line(Para) below end Code block or Image block

### Why is it needed?

Currently user is not able to type or add new block below code or image block, [Issue](https://strapi-inc.atlassian.net/browse/CS-261) 

### How to test it?

User can now add new block or type below the end code/image block.

### CODE REVIEW

File: packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js [hunk 1]
```diff
@@ -202,6 +202,27 @@ ImageDialog.propTypes = {
   handleClose: PropTypes.func.isRequired,
 };
 
+const isLastBlockImageORCode = (editor) => {
```
Comment (remidej) [inline packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js:205]: just a casing fix
```suggestion
const isLastBlockImageOrCode = (editor) => {
```

File: packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js [hunk 2]
```diff
@@ -202,6 +202,27 @@ ImageDialog.propTypes = {
...
+    at: selection,
+    match: (n) => n.type === 'image' || n.type === 'code',
+  });
+
+  if (currentImageORCodeBlock) {
+    const [, currentNodePath] = currentImageORCodeBlock;
+
+    const nodeAfter = Editor.after(editor, currentNodePath);
+
+    return !nodeAfter;
```
Comment (remidej) [inline packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js:220]: Just for readability could you return `!Boolean(nodeAfter)`? to make it obvious we're returning a boolean

File: packages/core/admin/admin/src/content-manager/components/BlocksEditor/hooks/useBlocksStore.js [hunk 3]
```diff
@@ -161,7 +161,7 @@ Image.propTypes = {
       alternativeText: PropTypes.string,
       width: PropTypes.number,
       height: PropTypes.number,
-    }).isRequired,
+    }),
```
Comment (remidej) [inline packages/core/admin/admin/src/content-manager/components/BlocksEditor/hooks/useBlocksStore.js:164]: Why is it no longer required?

**Review by remidej — COMMENTED** [review]
Could you also add some test cases?

File: packages/core/admin/admin/src/content-manager/components/BlocksEditor/hooks/useBlocksStore.js (hunk 3 above)
Comment (madhurisandbhor) [reply packages/core/admin/admin/src/content-manager/components/BlocksEditor/hooks/useBlocksStore.js:164]: Actually when there is a empty node or text node, and user tries to select image option, that time schema becomes as below, so this required condition violates. I have also added a condition when `!element.image return null` inside <Image /> element.
`
{
type: 'image',
children: [
   { 
      type: 'text', 
      text: ''
   }
]
}
`

**Review by remidej — COMMENTED** [review]
Could you add the same test for images? If it's harder because of the media library modal we could also add it after the alpha but I think it would be needed

**Review by remidej — APPROVED** [review]
Did QA and works as expected. We probably should merge now otherwise it won't make it in the alpha release. But I would appreciate if further tests were added later

**Comment by madhurisandbhor** [comment]
> Could you add the same test for images? If it's harder because of the media library modal we could also add it after the alpha but I think it would be needed

no its not a problem, but its the same method check for both types, so I thought its sufficient to check for code as we already have image test case, but will extend it.

**Pull request merged**